src/core/thread/mle_router.cpp

```diff
@@ -1080,6 +1087,40 @@ Error MleRouter::HandleLinkAccept(RxInfo &aRxInfo, bool aRequest)
     return error;
 }
 
+bool MleRouter::DetectAsymmetricLinks(const RouteTlv &aRouteTlv, RxInfo &aRxInfo)
```

Comment:
Thanks @sarveshkumarv3 for submitting this PR.

A couple of points:

- We decided to apply a random delay to the unicast Advertisement transmission.
- I see you added a new method, `DetectAsymmetricLinks()`, for this. It would be easier to check for this in `UpdateRoutes()`, where we already iterate over entries in a received `RouteTlv`.

I actually had a branch with an implementation of this (without a random delay) a while back. You may have noticed that more recently, I have been adding new mechanisms in `Mle` to allow us to schedule (with a random delay) any MLE message type easily. With this already merged, I rebased the older branch and updated it to apply a random delay and submitted it as [https://github.com/openthread/openthread/pull/10906](https://github.com/openthread/openthread/pull/10906).

Please check it out. I think it is smaller/simpler code. What do you think?

Reply:
sure, thanks!